bccsp/pkcs11/pkcs11.go

```diff
@@ -374,7 +376,7 @@ func (csp *Provider) getECKey(ski []byte) (pubKey *ecdsa.PublicKey, isPriv bool,
 	if err != nil {
 		return nil, false, err
 	}
-	defer csp.returnSession(session)
+	defer func() { csp.handleSessionReturn(err, session) }()
```

Comment:
Another option that might be cleaner, is that the **csp.getSession()** returns a cleanup callback according to the error status- if the error matches the regexp, the cleanup callback is a no-op. otherwise, it is the **closeSesson**.